src/helper/waylandsocketwatcher.cpp

```diff
@@ -28,18 +28,20 @@ namespace SDDM {
 WaylandSocketWatcher::WaylandSocketWatcher(QObject *parent )
     : QObject(parent)
     , m_runtimeDir(QDir(QStandardPaths::writableLocation(QStandardPaths::RuntimeLocation)))
-    , m_socketPath(m_runtimeDir.absoluteFilePath(QLatin1String("wayland-0")))
+    , m_socketName(QLatin1String("wayland-?"))
 {
+    m_runtimeDir.setFilter(QDir::Files | QDir::System);
+    m_runtimeDir.setNameFilters(QStringList() << m_socketName);
 }
 
 WaylandSocketWatcher::Status WaylandSocketWatcher::status() const
 {
     return m_status;
 }
 
-QString WaylandSocketWatcher::socketPath() const
```

Comment:
Let's just remove this method entirely? Nobody is calling socketPath anyway.

Reply:
`WaylandHelper::startGreeter` is calling `WaylandSocketWatcher::socketName` to get the `WAYLAND_DISPLAY` value. That's actually important, as the Wayland socket has to be explicitly passed to the greeter if it's not `wayland-0`.